src/devices/Dhtxx/DhtBase.cs

```diff
@@ -20,12 +21,13 @@ namespace Iot.Device.DHTxx
     [Interface("Temperature and Humidity Sensor DHTxx")]
     public abstract class DhtBase : IDisposable
     {
+        private const int BufferSize = 5;
         private readonly CommunicationProtocol _protocol;
 
         /// <summary>
         /// Read buffer
         /// </summary>
-        private ValueArray<byte> _readBuff = new ValueArray<byte>(5);
+        private byte[] _readBuff = new byte[BufferSize];
```

Comment:
Why do you need to initialize it here? Looks to me that everywhere it is used it gets initialized again anyway.

Reply:
If you set `MinTimeBetweenReads` to a value that will trigger the early-out in `ReadData` before you read from the sensor for the first time, you would get a `null` result handed back unless the buffer is initialized here.

Reply:
Yea, the main reason for the initialization here is just to avoid having to make the field nullable.